### PR TITLE
docs:  mandates and payments routes

### PR DESCRIPTION
**Pull request by Sangamesh26**
## Type of Change

- [x] Enhancement

## Description

Docs for payments and mandates routes

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ pub struct PaymentsRequest {
     /// Provide a reference to a stored payment method
     #[schema(example = "187282ab-40ef-47a9-9206-5099ba31e432")]
     pub payment_token: Option<String>,
-
+    /// This is used when payment is confirm and the card is not saved
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:99]:
```suggestion
    /// This is used when payment is to be confirmed and the card is not saved
```

File: crates/router/src/routes/mandates.rs [hunk 2]
```diff
@@ -8,6 +8,20 @@ use crate::{
     types::api::mandates,
 };
 
+/// Mandates - Get mandates
+///
+/// To get the mandates
```
Comment (SanchithHegde) [inline crates/router/src/routes/mandates.rs:13]:
```suggestion
/// Retrieve a mandate given the mandate ID.
```

File: crates/router/src/routes/mandates.rs [hunk 3]
```diff
@@ -28,6 +42,20 @@ pub async fn get_mandate(
     .await
 }
 
+/// Mandates - Revoke mandate
+///
+/// To revoke the mandate
```
Comment (SanchithHegde) [inline crates/router/src/routes/mandates.rs:47]:
```suggestion
/// Revoke a mandate given the mandate ID.
```

File: crates/router/src/routes/mandates.rs [hunk 4]
```diff
@@ -8,6 +8,20 @@ use crate::{
...
+///
+/// To get the mandates
+#[utoipa::path(
+    get,
+    path = "/mandates/{mandate_id}",
+    params(
+        ("mandate_id" = String, Path, description = "The identifier for mandate")
+    ),
+    responses(
+        (status = 200, description = "Get mandates", body = MandateResponse),
```
Comment (SanchithHegde) [inline crates/router/src/routes/mandates.rs:21]:
```suggestion
        (status = 200, description = "The mandate was retrieved successfully", body = MandateResponse),
```

File: crates/router/src/routes/mandates.rs [hunk 5]
```diff
@@ -28,6 +42,20 @@ pub async fn get_mandate(
...
+///
+/// To revoke the mandate
+#[utoipa::path(
+    post,
+    path = "/mandates/revoke/{mandate_id}",
+    params(
+        ("mandate_id" = String, Path, description = "The identifier for mandate")
+    ),
+    responses(
+        (status = 200, description = "Revoke mandates", body = MandateRevokedResponse),
```
Comment (SanchithHegde) [inline crates/router/src/routes/mandates.rs:55]:
```suggestion
        (status = 200, description = "The mandate was revoked successfully", body = MandateRevokedResponse),
```

File: crates/router/src/routes/payments.rs [hunk 6]
```diff
@@ -52,6 +52,22 @@ pub async fn payments_create(
     .await
 }
 
+/// Payments - Start
+///
+/// The entry point for payment which has redirection flow and it redirects to the authentication page
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:57]:
```suggestion
/// The entry point for a payment which involves the redirection flow. This redirects the user to the authentication page.
```

File: crates/router/src/routes/payments.rs [hunk 7]
```diff
@@ -52,6 +52,22 @@ pub async fn payments_create(
...
+#[utoipa::path(
+    get,
+    path = "/payments/start/{payment_id}/{merchant_id}/{attempt_id}",
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment"),
+        ("merchant_id" = String, Path, description = "The identifier for merchant"),
+        ("attempt_id" = String, Path, description = "The identifier for transaction")
+    ),
+    responses(
+        (status = 200, description = "Redirects to the authentication page", body = PaymentsResponse),
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:67]: Shouldn't this be a 30X status code?

File: crates/router/src/routes/payments.rs [hunk 8]
```diff
@@ -122,6 +153,21 @@ pub async fn payments_retrieve(
...
+#[utoipa::path(
+    post,
+    path = "/payments/{payment_id}",
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment")
+    ),
+    request_body=PaymentsRequest,
+    responses(
+        (status = 200, description = "Payment updated", body = PaymentsResponse),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:168]:
```suggestion
        (status = 400, description = "Missing mandatory fields")
```

File: crates/router/src/routes/payments.rs [hunk 9]
```diff
@@ -163,6 +209,21 @@ pub async fn payments_update(
...
+#[utoipa::path(
+    post,
+    path = "/payments/{payment_id}/confirm",
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment")
+    ),
+    request_body=PaymentsRequest,
+    responses(
+        (status = 200, description = "Payment confirmed", body = PaymentsResponse),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:224]:
```suggestion
        (status = 400, description = "Missing mandatory fields")
```

File: crates/router/src/routes/payments.rs [hunk 10]
```diff
@@ -205,6 +266,21 @@ pub async fn payments_confirm(
     .await
 }
 
+/// Payments - Capture
+///
+/// To capture payments for a currently uncaptured Payment with its status as requires_capture
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:271]:
```suggestion
/// To capture payments for a currently uncaptured payment
```

@bernard-eugine What do you think, do we need the `with its status as requires_capture` part again?

File: crates/router/src/routes/payments.rs [hunk 11]
```diff
@@ -205,6 +266,21 @@ pub async fn payments_confirm(
...
+#[utoipa::path(
+    post,
+    path = "/payments/{payment_id}/capture",
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment")
+    ),
+    request_body=PaymentsCaptureRequest,
+    responses(
+        (status = 200, description = "Payment captured", body = PaymentsResponse),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:281]:
```suggestion
        (status = 400, description = "Missing mandatory fields")
```

File: crates/router/src/routes/payments.rs [hunk 12]
```diff
@@ -237,6 +313,18 @@ pub async fn payments_capture(
...
+/// Payments - Session token
+///
+/// To create the session object or to get session token for wallets
+#[utoipa::path(
+    post,
+    path = "/payments/session_tokens",
+    request_body=PaymentsSessionRequest,
+    responses(
+        (status = 200, description = "Payment session object created or get session token from wallets", body = PaymentsSessionResponse),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:325]:
```suggestion
        (status = 200, description = "Payment session object created or session token was retrieved from wallets", body = PaymentsSessionResponse),
        (status = 400, description = "Missing mandatory fields")
```

File: crates/router/src/routes/payments.rs [hunk 13]
```diff
@@ -270,8 +358,24 @@ pub async fn payments_connector_session(
...
+    post,
+    path = "/payments/{payment_id}/{merchant_id}/response/{connector}",
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment"),
+        ("merchant_id" = String, Path, description = "The identifier for merchant"),
+        ("connector" = String, Path, description = "The name of the connector")
+    ),
+    responses(
+        (status = 200, description = "Recieved payment redirect response"),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:374]:
```suggestion
        (status = 200, description = "Received payment redirect response"),
        (status = 400, description = "Missing mandatory fields")
```

Is this supposed to be a 200 status code or 30X?

File: crates/router/src/routes/payments.rs [hunk 14]
```diff
@@ -302,6 +406,21 @@ pub async fn payments_response(
...
+#[utoipa::path(
+    post,
+    path = "/payments/{payment_id}/cancel",
+    request_body=PaymentsCancelRequest,
+    params(
+        ("payment_id" = String, Path, description = "The identifier for payment")
+    ),
+    responses(
+        (status = 200, description = "Payment canceled"),
+        (status = 400, description = "Missing Mandatory fields")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:421]:
```suggestion
        (status = 400, description = "Missing mandatory fields")
```

File: crates/router/src/routes/payments.rs [hunk 15]
```diff
@@ -333,6 +452,28 @@ pub async fn payments_cancel(
...
+    params(
+        ("customer_id" = String, Query, description = "The identifier for the customer"),
+        ("starting_after" = String, Query, description = "A cursor for use in pagination, fetch the next list after some object"),
+        ("ending_before" = String, Query, description = "A cursor for use in pagination, fetch the previous list before some object"),
+        ("limit" = i64, Query, description = "Limit on the number of objects to return"),
+        ("created" = PrimitiveDateTime, Query, description = "The time at which refund is created"),
+        ("created_lt" = PrimitiveDateTime, Query, description = "Time less than the refund created time"),
+        ("created_gt" = PrimitiveDateTime, Query, description = "Time greater than the refund created time"),
+        ("created_lte" = PrimitiveDateTime, Query, description = "Time less than or equals to the refund created time"),
+        ("created_gte" = PrimitiveDateTime, Query, description = "Time greater than or equals to the refund created time")
```
Comment (SanchithHegde) [inline crates/router/src/routes/payments.rs:470]: Replace refund with payment

**Review by SanchithHegde — CHANGES_REQUESTED** [review]

**Review by SanchithHegde — DISMISSED** [review]

**Review by jarnura — APPROVED** [review]
LGTM

**Comment by Sangamesh26** [comment]
Merged into #453

**Pull request closed without merging**